pkg/u8g2/Makefile.u8g2_csrc

```diff
@@ -1,6 +1,7 @@
 MODULE = u8g2_csrc
 
 CFLAGS += -Wno-overlength-strings
+CFLAGS += -Wno-unused-variable
```

Comment:
Please add documentation on why this flag is added (I know the previous does not either, but let's not repeat mistakes ;-))

Reply:
Ping @gschorcht?

Reply:
Hm, not really clear what comment could say more than this option.

Reply:
Why it was added, what needs to happen to remove the flag, maybe mention `NDEBUG`. ;-)

Reply:
I agree with @gschorcht, there's no real need for a comment here. And this Makefile is building code from an external package, so I wouldn't care that much that there's an unused variable.

@gschorcht, please rebase on top of latest master so all new CI checks are triggered.